Homework6

senvarin opened the pull request:
Есть вопрос по поводу теста App. В самом приложении логин и пароль задаются через переменные окружения в application.properties и если не задать их для теста, то он падает. Я задал данные от рельное БД, но думаю, что так делать не стоит. Что туда нужно подставить для тестирования?

senvarin committed:
Added creation of Secrutiy Audit via aspect

senvarin committed:
Added search DTO for entries and assessments

senvarin committed:
Tests

Added test for audit controller, service, repository and DTO validation
tests and fixed old ones

senvarin committed:
Renamed package 'controller' -> 'api'

RavenVSS commented on src/main/java/org/kaoden/ws/homework/api/assessment/AssessmentController.java:
```diff
@@ -40,10 +41,9 @@ public AssessmentDto create(@Valid @RequestBody CreateAssessmentDto assessment)
     @Operation(description = "Getting all assessments for the entry")
-    public List<AssessmentDto> getAll(@RequestParam(name = "entryId") Long entryId,
-                                      @RequestParam(required = false) Integer value,
+    public List<AssessmentDto> getAll(@Valid @RequestBody SearchAssessmentDto searchDto,
```
Убери `@RequestBody`, его используем только для POST запрсов  
Если ничего не указывать из аннотаций, то поля в Dto будут восприниматься как `@RequestParam`

RavenVSS commented on src/main/java/org/kaoden/ws/homework/api/audit/SecurityAuditController.java:
```diff
@@ -0,0 +1,39 @@
+    SecurityAuditMapper mapper;
+
+    @GetMapping("all")
+    public List<SecurityAuditDto> getAll(@Valid @RequestBody SearchSecurityAuditDto searchDto,
```
Аналогично с `@RequestBody`

RavenVSS commented on src/main/java/org/kaoden/ws/homework/service/audit/SecurityAuditServiceImpl.java:
```diff
@@ -0,0 +1,49 @@
+    public Page<SecurityAudit> getAll(SearchSecurityAuditArgument argument, Pageable pageable) {
+        String info = argument.getInfo();
+        if (info == null || info.isBlank()) {
+            throw new SearchSecurityAuditException("No info for searching");
```
Если ничего не указано в параметрах поиска, нужно выводить всё.  
В данном случае можно попробовать просто пустую строку передавать:
```
String info = argument.getInfo() == null ? "" : argument.getInfo();
or
if (info == null) info = "";
```
По идее так должно сработать

RavenVSS commented on src/test/java/org/kaoden/ws/homework/AppTest.java:
```diff
@@ -3,7 +3,9 @@
-@SpringBootTest
+@SpringBootTest(
+        properties = {"DB_LOGIN=vladislav", "DB_PASSWORD=admin"}
+)
```
Замени на аннотацию `@EnablePostgresIntegrationTest`, получается она делает тоже самое что у и у тебя, только + запускает докер контейнер с базой итп

Если нормально сработает, можешь во всех IT тестах эту аннотацию проставить

Скорей всего ещё придётся в pom добавить:
```
<dependency>
    <groupId>com.jupiter-tools</groupId>
    <artifactId>spring-test-postgres</artifactId>
    <version>0.4.1</version>
    <scope>test</scope>
</dependency>
```
Потом лишние зависимости можно будет удалить через плагин для IDE - Maven Helper

senvarin replied:
Я как раз таки не очень хочу использовать `@EnablePostgresIntegrationTest`, по той причине, что придется понижать версию Spring до 2.7 или около того, а вместе с понижением версии, я смотрел, надо будте некоторые места сильно менять. Так например в 2.7 нет  `MethodArgumentNotValidExceptio`, который я использовал в `ApiExceptinoHandler` и так еще некоторые места. Поэтому я хотел знать, как лучше поступить с DB_LOGIN и DB_PASSWORD, если это возможно, без использования  `@EnablePostgresIntegrationTest`?

RavenVSS replied:
Смотри сам, тогда судя по всему достаточно добавить `@Testcontainers` и
```
@Container
@ServiceConnection
static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:15");
```
Ты это делал в другом IT тесте. Должно сработать

senvarin committed:
Refactoring

Removing the RequestBody annotation from SearchDTO, minor changes SecurityAudit, change method name of SecurityAudit
repository and adding minor changes into tests

RavenVSS commented on src/main/java/org/kaoden/ws/homework/service/audit/SecurityAuditServiceImpl.java:
```diff
@@ -32,11 +31,11 @@ public SecurityAudit create(CreateSecurityAuditArgument argument) {
+            return repository.findAll(pageable);
         }
-        return repository.findAllByInfoContainingIgnoreCase(info, pageable);
+        return repository.findSecurityAuditsByInfoContainingIgnoreCase(info, pageable);
```
Может тогда лучше просто `StringUtils.isBlank(argument.getInfo())` в условии задать, чтобы пустую строку не присваивать

Я думал сюда пустую строку передавать, но тогда он null значения не будет выводить. В идеале сделать через QueryDsl, но пока так пойдёт

senvarin committed:
Minor refactoring
